salesforce/read_test.go

```diff
@@ -0,0 +1,214 @@
+package salesforce
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/test/utils/mockutils"
+	"github.com/go-test/deep"
+)
+
+func TestRead(t *testing.T) { //nolint:funlen,gocognit,cyclop,maintidx
+	t.Parallel()
+
+	responseUnknownObject := mockutils.DataFromFile(t, "unknown-object.json")
+	responseLeadsFirstPage := mockutils.DataFromFile(t, "read-list-leads.json")
+	responseListContacts := mockutils.DataFromFile(t, "read-list-contacts.json")
+
+	tests := []struct {
+		name         string
+		input        common.ReadParams
```

Comment:
I'm realizing that we're not doing any validation that ReadParams has a `ObjectName` field, which should be a required field. Once that validation exists, these test cases should be updated to include an ObjectName in the input, and there should be one more test case added to ensure ObjectName is required.

Reply:
Yes, this was not enforced for any connector Read operation unlike the Write method.